packages/neo-one-smart-contract-compiler/src/compile/syscalls.ts

```diff
@@ -694,37 +696,53 @@ export class SysCallArgument {
     return `${this.name}: ${this.type.toSignature()}`;
   }
 }
+export type SysCallHandleCallFnx = (sb: ScriptBuilder, node: CallExpression, options: VisitOptions) => void;
+export type SysCallToDeclarationFnx = () => string;
 
 export interface SysCall {
   readonly name: string;
-  readonly toDeclaration: () => string;
-  readonly handleCall: (sb: ScriptBuilder, node: CallExpression, options: VisitOptions) => void;
+  readonly toDeclaration: SysCallToDeclarationFnx;
+  readonly handleCall: SysCallHandleCallFnx;
 }
 
 export interface SimpleSysCallOptions {
   readonly name: SysCallName;
   readonly args?: ReadonlyArray<SysCallArgument>;
+  readonly toDeclaration?: SysCallToDeclarationFnx | undefined;
+  readonly handleCall?: SysCallHandleCallFnx | undefined;
   readonly returnType?: SysCallType;
 }
 
 export class SimpleSysCall implements SysCall {
   public readonly name: SysCallName;
   private readonly args: ReadonlyArray<SysCallArgument>;
   private readonly returnType: SysCallType;
+  private readonly _toDeclaration: SysCallToDeclarationFnx | undefined;
+  private readonly _handleCall: SysCallHandleCallFnx | undefined;
 
-  public constructor({ name, args = [], returnType = VoidValue }: SimpleSysCallOptions) {
+  public constructor({ name, args = [], returnType = VoidValue, toDeclaration, handleCall }: SimpleSysCallOptions) {
```

Comment:
Re: offline conversation - pass in just `declaration` as a string and revert the other changes.